packages/ai/src/ai/account/store_providers/filesystem.py

```
@@ -292,6 +317,72 @@ async def list_files(self, prefix: str = '') -> list:
 
+    # =========================================================================
+    # Handle-Based I/O
+    # =========================================================================
+
+    async def open_write(self, filename: str) -> dict:
+        """Open a file for writing. Returns context with aiofiles handle."""
+        try:
+            full_path = self._get_full_path(filename)
+            full_path.parent.mkdir(parents=True, exist_ok=True)
+            f = await aiofiles.open(full_path, 'wb')
+            return {'context': {'file': f, 'path': full_path}}
```

Ruff (0.15.7)

[warning] Consider moving this statement to an `else` block

(TRY300)